bindings/java/java/src/main/java/org/ethereum/evmc/Host.java

```diff
@@ -16,6 +16,13 @@
  * <p>The set of all callback functions expected by VM instances.
  */
 final class Host {
+  static private ByteBuffer ensureDirectBuffer(ByteBuffer input) {
+    // Reallocate if needed.
+    if (!input.isDirect())
```

Comment:
No idea what it does. Is this even tested?

Reply:
Not all of them are, but `get/set_storage` is. This helper enforces that the `ByteBuffer` is in a format ("direct buffeR") which can be queried from JNI (e.g. the `GetDirectBuffer` helper in C won't assert).